main/guides/getting-started/how-to-use-ui-kit.md

```diff
@@ -0,0 +1,147 @@
+
+## Reading Contract Data (vstorage)
+The UI Kit includes utilities for reading data from vstorage, the virtual storage system used by Agoric contracts. The `makeAgoricChainStorageWatcher` function allows you to subscribe to updates to vstorage and efficiently retrieve data using the Agoric RPC API.
+
+```js
+import {
+  makeAgoricChainStorageWatcher,
+  AgoricChainStoragePathKind as Kind,
+} from '@agoric/rpc';
+
+const watcher = makeAgoricChainStorageWatcher(restApi, chainName);
+
+// Watch vstorage children at a given node.
+const stopWatching = watcher.watchLatest<string[]>(
+  [Kind.Children, 'published.vaultFactory.managers'],
+  managerIds => {
+    console.log('Got vault manager IDs:', managerIds);
+  },
+);
+
+// Stop watching.
+stopWatching();
+
+// Watch vstorage data at a given node.
+watcher.watchLatest<Brands>(
+  [Kind.Data, 'published.agoricNames.brand'],
+  brands => {
+    console.log('Do something with the brands');
+  },
+);
+```
+
+### Connecting to User's Account (Keplr)
+The UI Kit provides the `makeAgoricWalletConnection` function to connect to the user's wallet using Keplr, a browser extension for interacting with Agoric and Cosmos-based blockchains. This function allows you to retrieve information about the user's purses (token accounts) and other account-related data.
+
+```js
+import { subscribeLatest } from '@agoric/notifier';
+import { makeAgoricChainStorageWatcher } from '@agoric/rpc';
+import { makeAgoricWalletConnection } from '@agoric/web-components';
+
+const watcher = makeAgoricChainStorageWatcher(apiAddr, chainName);
+const connection = await makeAgoricWalletConnection(watcher, rpcAddr);
+const { pursesNotifier, publicSubscribersNotifier } = chainConnection;
+
+for await (const purses of subscribeLatest(pursesNotifier)) {
+  console.log('Got purses:', purses);
+}
+```
+
+### Using a Custom Signer
+If you're using a custom signer instead of Keplr, you can provide your own signer implementation when creating the `AgoricWalletConnection`. Here's an example of setting up a custom signer using the `@cosmjs/stargate` library:
+
+```js
+import {
+  agoricRegistryTypes,
+  agoricConverters,
+  makeAgoricWalletConnection,
+} from '@agoric/web-components';
+import { Registry } from '@cosmjs/proto-signing';
+import {
+  AminoTypes,
+  defaultRegistryTypes,
+  createBankAminoConverters,
+  createAuthzAminoConverters,
+} from '@cosmjs/stargate';
+
+// ...
+
+const signingStargateClient = await SigningStargateClient.connectWithSigner(
+  rpcEndpoint, // RPC endpoint to use
+  customAminoSigner, // E.g. window.getOfflineSignerOnlyAmino(chainId)
+  {
+    aminoTypes: new AminoTypes({
+      ...agoricConverters,
+      ...createBankAminoConverters(),
+      ...createAuthzAminoConverters(),
+    }),
+    registry: new Registry([...defaultRegistryTypes, ...agoricRegistryTypes]),
+  },
```

Comment:
https://github.com/Agoric/ui-kit/pull/82/files#diff-9c512225baf4e3651b0ddbff72b0de5ed0304c4a9e997f0f8926583af3499c18 shows how to use `getSigningAgoricClientOptions()` to simplify this. Though I don''t know how it interacts with `custoMAminoSigner`. I bet @samsiegart does.

Reply:
I think `customAminoSigner` maps to `offlineSigner` fine in the above example. But, I don't know if we want to use `getSigningAgoricClientOptions` just yet because of https://github.com/Agoric/dapp-inter/issues/252